File: lifeFull/SpringBootLifeSamadhan/src/main/java/com/lifesamadhan/api/model/ServiceRequest.java
package com.lifesamadhan.api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "service_requests")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ServiceRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "CustomerId is required")
    @Min(value = 1, message = "CustomerId must be a valid value")
    @Column(name = "customer_id", nullable = false)
    private Long customerId;

    @NotNull(message = "ServiceId is required")
    @Min(value = 1, message = "ServiceId must be a valid value")
    @Column(name = "service_id", nullable = false)
    private Long serviceId;

    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "provider_id")
    private Long providerId;

    @Pattern(regexp = "^(PENDING|ASSIGNED|IN_PROGRESS|STARTED|COMPLETED|CANCELLED)$", message = "Status must be PENDING, ASSIGNED, IN_PROGRESS, STARTED, COMPLETED or CANCELLED")
    @Builder.Default
    @Column(nullable = false)
    private String status = "PENDING";

    @Column(name = "service_address")
    private String serviceAddress;

    @Column(name = "scheduled_date")
    private LocalDateTime scheduledDate;

    @Column(name = "amount")
    private Double amount;

    @Column(name = "payment_status")
    @Builder.Default
    private String paymentStatus = "PENDING";

    @Column(name = "paid_amount")
    @Builder.Default
    private Double paidAmount = 0.0;

    @Column(name = "completion_date")
    private LocalDateTime completionDate;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", insertable = false, updatable = false)
    private CustomerProfile customer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "service_id", insertable = false, updatable = false)
    private Service service;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "location_id", insertable = false, updatable = false)
    private Location location;

    @OneToMany(mappedBy = "request", cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    @com.fasterxml.jackson.annotation.JsonIgnore
    private java.util.List<ServiceAssignment> assignments;
}

File: LifeSamadhan/Backend/Models/ServiceRequest.cs

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LifeSamadhan.API.Models
{
    [Table("servicerequests")]
    public class ServiceRequest
    {
        [Key]
        [Column("Id")]
        public long Id { get; set; }

        [Column("CustomerId")]
        public long CustomerId { get; set; }

        [Column("ServiceId")]
        public long? ServiceId { get; set; }

        [Column("ServiceCategoryId")]
        public long? ServiceCategoryId { get; set; }

        [Column("LocationId")]
        public long LocationId { get; set; }

        [Column("ProviderId")]
        public long? ProviderId { get; set; }

        [Column("ServiceAddress")]
        public string ServiceAddress { get; set; } = string.Empty;

        [Column("ScheduledDate")]
        public DateTime ScheduledDate { get; set; }

        [Column("Status")]
        public string Status { get; set; } = "REQUESTED";

        [Column("OTP")]
        public string? OTP { get; set; } 

        [Column("CompletionDate")]
        public DateTime? CompletionDate { get; set; }

        [Column("PaymentStatus")]
        public string PaymentStatus { get; set; } = "PENDING";

        [Column("Amount")]
        public decimal Amount { get; set; }

        [Column("CreatedAt")]
        public DateTime CreatedAt { get; set; } = DateTime.Now;

        
        [ForeignKey("CustomerId")]
        public User? Customer { get; set; }

        [ForeignKey("ServiceId")]
        public Service? Service { get; set; }

        [ForeignKey("ServiceCategoryId")]
        public ServiceCategory? ServiceCategory { get; set; }

        [ForeignKey("ProviderId")]
        public ServiceProvider? Provider { get; set; }

        [ForeignKey("LocationId")]
        public Location? Location { get; set; }
    }
}